tests/remote_resource/test_requests.py

```diff
@@ -0,0 +1,257 @@
+                }
+            }
+        }
+
+        resource = RemoteResourceWrapper(data['data'])
+
+        self.assertEqual(resource.title, "Omakase")
+        self.assertEqual(resource.authors[0].type, "People")
+        self.assertEqual(resource.authors[0].id, "9")
+
+    def test_remote_resource_list_wrapper_one_relationship(self):
+        data = {
+            "data": [
+                {
+                    "type": "articles",
+                    "id": "1",
+                    "attributes": {
+                        "title": "Omakase"
+                    },
+                    "relationships": {
+                        "author": {
+                            "links": {
+                                "self": "/articles/1/relationships/author",
+                                "related": "/articles/1/author"
+                            },
+                            "data": {"type": "People", "id": "9"}
+                        }
+                    }
+                }
+            ]
+        }
+
+        resources = RemoteResourceListWrapper(data['data'])
+
+        self.assertEqual(resources[0].type, 'articles')
+        self.assertEqual(resources[0].id, '1')
+        self.assertEqual(resources[0].title, 'Omakase')
+        self.assertEqual(resources[0].author.type, 'People')
+        self.assertEqual(resources[0].author.id, '9')
+
+    def test_remote_resource_list_wrapper_multiple_relationship(self):
+        data = {
+            "data": [
+                {
+                    "type": "articles",
+                    "id": "1",
+                    "attributes": {
+                        "title": "Omakase"
+                    },
+                    "relationships": {
+                        "authors": {
+                            "links": {
+                                "self": "/articles/1/relationships/authors",
+                                "related": "/articles/1/authors"
+                            },
+                            "data": [
+                                {"type": "People", "id": "9"},
+                            ]
+                        }
+                    }
+                }
+            ]
+        }
+
+        resources = RemoteResourceListWrapper(data['data'])
+
+        self.assertEqual(resources[0].type, 'articles')
+        self.assertEqual(resources[0].id, '1')
+        self.assertEqual(resources[0].title, 'Omakase')
+        self.assertEqual(resources[0].authors[0].type, 'People')
+        self.assertEqual(resources[0].authors[0].id, '9')
+
+
+@mock.patch('zc_common.remote_resource.request.requests')
+@mock.patch('zc_common.remote_resource.request.zc_settings')
+class RouteRetrievalTestCase(TestCase):
+
+    def setUp(self):
+        self.gateway_root_path = 'http://mp-gateway.zerocater.com'
+        self.mappings = {
+            '/users/{id}': {
+                'domain': 'https://mp-users.zerocater.com',
+                'resource_type': 'User'
+            }
```

Comment:
The gateway currently returns the trailing slash inside the curly braces, which allows us to append query parameters without the trailing slash: `{"/users{/id}":{"domain":"https://mp-users.herokuapp.com","resource_type":"User"}`